shotover/src/transforms/kafka/sink_cluster/mod.rs

```diff
@@ -2716,19 +2726,35 @@ impl KafkaSinkCluster {
                     .unwrap()
                     .rack
                     .as_ref();
-                let shotover_node = self
-                    .shotover_nodes
-                    .iter()
-                    .find(|shotover_node| {
-                        coordinator_rack
-                            .map(|rack| rack == &shotover_node.rack)
-                            .unwrap_or(true)
-                    })
-                    .unwrap();
 
-                find_coordinator.host = shotover_node.address.host.clone();
-                find_coordinator.port = shotover_node.address.port;
-                find_coordinator.node_id = shotover_node.broker_id;
+                let hash = hash_str_bytes(request.key.clone());
+
+                let (shotover_nodes_in_rack, shotover_nodes_out_of_rack): (
+                    Vec<&ShotoverNode>,
+                    Vec<&ShotoverNode>,
+                ) = up_shotover_nodes.iter().partition(|shotover_node| {
+                    coordinator_rack
+                        .map(|rack| rack == &shotover_node.rack)
+                        .unwrap_or(true)
+                });
+
+                let shotover_node: Option<&&ShotoverNode> = shotover_nodes_in_rack
+                    .get(hash % shotover_nodes_in_rack.len())
+                    .or_else(|| {
+                        shotover_nodes_out_of_rack.get(hash % shotover_nodes_out_of_rack.len())
+                    });
+
+                match shotover_node {
+                    Some(shotover_node) => {
+                        find_coordinator.host = shotover_node.address.host.clone();
+                        find_coordinator.port = shotover_node.address.port;
+                        find_coordinator.node_id = shotover_node.broker_id;
+                    }
+                    None => {
+                        // Return the COORDINATOR_NOT_AVAILABLE error if all shotover nodes are down
+                        find_coordinator.error_code = 15;
```

Comment:
Its impossible for every shotover node to be down since the current shotover node must be up, otherwise this code wouldnt be running

Reply:
yeah removed this COORDINATOR_NOT_AVAILABLE and updated the comment